pylops_mpi/basicoperators/FirstDerivative.py

```diff
@@ -0,0 +1,285 @@
+from typing import Callable
+import numpy as np
+
+from pylops.utils import DTypeLike
+
+from pylops_mpi import (
+    DistributedArray,
+    MPILinearOperator,
+    Partition
+)
+
+
+class MPIFirstDerivative(MPILinearOperator):
+    r"""MPI First Derivative
+
+    Apply a first derivative using a multiple-point stencil finite-difference
+    approximation with :class:`pylops_mpi.DistributedArray`.
+
+    Parameters
+    ----------
+    dims : :obj:`int`
+        Number of samples for model and data vector.
+    sampling : :obj:`float`, optional
+        Sampling step :math:`\Delta x`.
+    kind : :obj:`str`, optional
+        Derivative kind (``forward``, ``centered``, or ``backward``).
+    edge : :obj:`bool`, optional
+        Use reduced order derivative at edges (``True``) or
+        ignore them (``False``). This is currently only available
+        for centered derivative.
+    order : :obj:`int`, optional
+        Derivative order (``3`` or ``5``). This is currently only available
+        for centered derivative.
+    dtype : :obj:`str`, optional
+        Type of elements in the input array.
+
+    Attributes
+    ----------
+    shape : :obj:`tuple`
+        Operator shape
+
+    Notes
+    -----
+    The MPIFirstDerivative operator applies a first derivative to a :class:`pylops_mpi.DistributedArray`
+    using either a first-order backward and forward stencil or a second or third ordered centered stencil.
+
+    When computing the first derivative using a :class:`pylops_mpi.DistributedArray`, a technique named **ghosting**
+    is employed, often in conjunction with MPI for communication between different processes. Ghosting involves
+    creating additional **"ghost cells"** around the boundary of each process's local data domain. These ghost cells
+    are replicated copies of the border cells from neighboring processes, these cells allow each process
+    to perform local computations without the need for explicit communication with other processes.
```

Comment:
We should also add a short comment that we currently do not handle edges, so this operator behaves like pylops equivalent with `edge=False`